src/core/functions/internal_functions/web/InternalModuleWeb.ts

```diff
@@ -80,4 +82,31 @@ export class InternalModuleWeb extends InternalModule {
             }
         };
     }
+
+    make_request(): (
+        url: string,
+        path?: string,
+    ) => Promise<string> {
+        return async (url: string, path?: string) => {
+            try {
+                const response = await this.getRequest(url);
+                const jsonData = await response.json();
+
+                if (path && jsonData) {
+                    return path.split('.').reduce((obj, key) => {
+                        if (obj && obj.hasOwnProperty(key)) {
+                            return obj[key];
+                        } else {
+                            throw new Error(`Path ${path} not found in the JSON response`);
+                        }
+                    }, jsonData);
+                }
+
+                return JSON.stringify(jsonData, null, 2);
```

Comment:
What are your thoughts on returning the raw object instead of stringifying it? That way a user could make a single request to get multiple pieces of data. For example, instead of doing this:

```js
Title: <% tp.web.request("https://jsonplaceholder.typicode.com/todos", "0.title") %>
Completed: <% tp.web.request("https://jsonplaceholder.typicode.com/todos", "0.completed") %>
```
A user could do this:
```js
<%* const { title, completed } = await tp.web.request("https://jsonplaceholder.typicode.com/todos", "0"); -%>
Title: <% title %>
Completed: <% completed %>
```

Reply:
Yep, I think that's much better.